agent/app/agent.go

```diff
@@ -997,29 +999,67 @@ func (agent *ecsAgent) startACSSession(
 	taskHandler *eventhandler.TaskHandler,
 	doctor *doctor.Doctor) int {
 
-	acsSession := acshandler.NewSession(
-		agent.ctx,
-		agent.cfg,
-		deregisterInstanceEventStream,
-		agent.containerInstanceARN,
-		agent.credentialProvider,
-		agent.dockerClient,
+	inactiveInstanceCB := func() {
+		// If the instance is inactive (i.e., was deregistered), send an event to the event stream
+		// for the same.
+		err := deregisterInstanceEventStream.WriteToEventStream(struct{}{})
+		if err != nil {
+			seelog.Debugf("Failed to write to deregister container instance event stream, err: %v", err)
+		}
+	}
+
+	dockerVersion, err := taskEngine.Version()
+	if err != nil {
+		if err != nil {
+			seelog.Debugf("Failed to get docker version from task engine, err: %v", err)
```

Comment:
Similar here, do we want `debug`? And what's the impact of missing docker version?

Reply:
> what's the impact of missing docker version?
The impact of missing docker version is dockerVersion key and value not being present in the ACS WebSocket URL. This matches the behavior of the current corresponding code here: https://github.com/aws/amazon-ecs-agent/blob/dev/agent/acs/handler/acs_handler.go#L374-L376

> Similar here, do we want debug?
Since missing dockerVersion is not critically blocking (i.e., ACS WebSocket URL can be generated without it), I decided to use log level DEBUG here. I don't have a strong preference for this though - do you think there is a more fitting log level to use here in this case?

Reply:
Maybe warn or error? I see the code in later part and I am indifferent, the only impact might just be adoption metrics. I would not think this will typically happen so maybe an error.

Reply:
Hmm got it, I'll change to level WARN in the next revision. Thanks for bringing this up!